chaste_codegen/chaste_model.py

```diff
@@ -415,7 +412,7 @@ def _get_equations_for_ionic_vars(self):
                                             == self._units.get_unit(unit_cap['units']).dimensionality
                                             and eq.lhs not in self._ionic_derivs]
                 equations = [eq.lhs for eq in equations]
-            desired_units_and_capacitance = unit_cap
+            desired_units_and_capacitance = unit_cap.copy()
```

Comment:
This is interesting!
Is `unit_cap` a pint.Unit?

Reply:
no it's a dictionary with a pint unit and some other information to decide how to handle units dimensionally equivalent to the unit. The things is that I change `desired_units_and_capacitance['units']` later down and that unit_cap points to a class level variable hence the interference

Reply:
Ah. Thanks!